### PR TITLE
Add basic syntax support for Racket

### PR DESCRIPTION
**Pull request by Eugleo**
As per #1513. It is based on the file for Scheme; I decided not to use the advanced regex file, and opted for the basic one instead.

### CODE REVIEW

**Comment by RunDevelopment** [comment]
Thanks for the implementation!

Now, there a few things we will have to do before we can accept this:

1. Add Racket to `components.json`. Otherwise, Prism doesn't know your language definition exists.
1. Run gulp via the `npx gulp` command. This will rebuild Prism and (among other things) add a minified version of your language definition.

It would also be great if you were to add a few [tests](https://prismjs.com/test-suite.html).

**Comment by Eugleo** [comment]
@RunDevelopment Sure, I wondered whether some "registration" of the new file will be needed. Done! 
I'm quite bussy at the moment, but rather than simply copying over the tests from Scheme, I'll reserve some time in the future to do some better ones. So, no tests now, if it's ok.

File: components.json [hunk 1]
```diff
@@ -614,6 +614,11 @@
 			"title": "R",
 			"owner": "Golmote"
 		},
+		"racket": {
+			"title": "Racket",
+			"owner": "Eugleo"
+		},
+
```
Comment (RunDevelopment) [inline components.json]: Why is this extra line here?

**Comment by RunDevelopment** [comment]
@Eugleo Sure, take your time for the tests 😄

Btw. Don't worry about the failed checks. It's not your fault.

File: components.json (hunk 1 above)
Comment (Eugleo) [reply components.json]: No idea. Will fix, sorry.

**Comment by Eugleo** [comment]
@RunDevelopment Should be ok now. Thanks for the heads up about the checks — I'm fairly new to GitHub and don't really understand these things 😃 Will you merge now or after the tests? I'm not sure when I'll be able to add them...

**Comment by RunDevelopment** [comment]
@Eugleo 
> Will you merge now or after the tests?

I'm sorry but I can't merge anything. A maintainer will have to do that. But before that, we will thoroughly review your language definition.

I will most likely review it in the next few days or so.

**Comment by RunDevelopment** [comment]
@Eugleo 
While using Scheme as the base for Racket is not a bad idea, please use the `Prism.languages.extend` method to do so.

Also, Racket supports some features which Scheme does not and they should be added as well (maybe not all of them).
I can really recommend [Beau­ti­ful Racket](https://beautifulracket.com/#explainers) which has short explainations on all Racket features many of which differ from Scheme. E.g. [name keyword arguments](https://beautifulracket.com/explainer/functions.html#input), [condinitionals with `[]`](https://beautifulracket.com/explainer/booleans-and-conditionals.html#conditionals), and more.

**Comment by Eugleo** [comment]
@RunDevelopment I know Racket well, but from the conversation in the issue I linked I got the feeling that it will be better to have a minimal implementation. But I agree with you; I'll work on a more complete grammar/regex definition for Racket and rework this PR.

**Comment by RunDevelopment** [comment]
@Eugleo 
It's totally fine to keep it minimalistic for now. We can always add more complex language features after we have base support.

It's just that some Racket features (like [conditionals with `[]`](https://beautifulracket.com/explainer/booleans-and-conditionals.html#conditionals)) will break Scheme highlighting.
I don't know Racket very well, but I think that we should cover commonly used features.

**Comment by mAAdhaTTah** [comment]
@Eugleo @RunDevelopment Is there still work to be done on this PR?

**Comment by mAAdhaTTah** [comment]
@Eugleo No worries! Just wanted to make sure we didn't leave you hanging.

**Pull request closed without merging**